### ISSUE TITLE
Regex matching commands

### ISSUE THREAD

**Issue by LecrisUT** [aklajnert/pytest-subprocess#154]
I want to mock specific calls to subprocess but the command inputs are varying, e.g. `cmake -S/path/to/generated_data -B/random/build_path`. I would thus like to use regex match these arguments and make sure other commands like `cmake --build` are not called.

**Comment by aklajnert** [comment]
There is no regex support at the moment, but perhaps you can make use of the non-exact command matching feature: https://pytest-subprocess.readthedocs.io/en/latest/usage.html#non-exact-command-matching

**Comment by LecrisUT** [comment]
I can partially use it, but I want to selectively mock these, i.e. `["cmake", "--build", fp.any(min=1)]` (succeed), `["cmake", re.compile(r"-S\S*"), re.comiple(r"-B\S*"), fp.any()]` (succeed), `["cmake", fp.any()]` (everything else must fail)

My suggestion here is to allow to pass it `re.compile` if we want to check regex. It would be nice to pass it only r-string, but I don't think that would tell us if it's regex or not.

**Comment by aklajnert** [comment]
I agree that regex may be helpful here. I'll try to add support for it soon, but I can't promise anything. Feel free to submit a PR with your proposal.
